src/main.js

```diff
@@ -1,131 +1,243 @@
+    // todo set environment variables and run setvars scripts, if enabled
+    // todo hide GNU linker on Windows so MSVC is found
+}
 
-  for (const component of components) {
-    const url = componentUrls[component]
-    if (!url) {
-      core.error(`Unknown oneapi component: ${component}`)
-      process.exit(1)
+async function installList(installPath, components) {
+    for (const component of components) {
+        console.log(`Installing component: ${component}`)
+        await install(installPath, component)
     }
-    key = key + ':' + path.parse(url).base
-  }
-  console.log(`Key ${key}`)
-
-  console.log('Restoring from cache')
-  const restoreKey = await cache.restoreCache(['/opt/intel/oneapi'], key)
-
-  if (restoreKey) {
-    console.log(`Restore succeeded: ${restoreKey}`)
-    // clear key so we will not try to save
-    key = ''
-    return true
-  } else {
-    console.log(`Restore failed: ${restoreKey}`)
-    return false
-  }
 }
 
-async function prune () {
-  if (!core.getBooleanInput('prune')) { return }
-
-  const dirs = ['/opt/intel/oneapi/compiler/latest/linux/compiler/lib/ia32_lin',
-    '/opt/intel/oneapi/compiler/latest/linux/bin/ia32',
-    '/opt/intel/oneapi/compiler/latest/linux/lib/emu',
-    '/opt/intel/oneapi/compiler/latest/linux/lib/oclfpga']
-
-  console.log('Pruning oneapi install')
-  await exec.exec('du', ['-sh', '/opt/intel/oneapi'])
-  // rmRF does not have root permission
-  await exec.exec('sudo', ['rm', '-rf'].concat(dirs))
-  await exec.exec('du', ['-sh', '/opt/intel/oneapi'])
-}
+function list(verbose) {
+    let components = [];
+    if (verbose)
+        console.log('Available components:')
+
+    for (let platform in componentUrls) {
+        for (let component in platform) {
+            if (verbose)
+                console.log(`  ${platform} ${component}: ${platform[component]}`)
+            components.push({
+                platform: platform,
+                component: component,
+                version: platform[component]
+            })
+        }
+    }
 
-async function install (component) {
-  const url = componentUrls[component]
-  console.log(`Installing ${component} from ${url}`)
-  const installerPath = await tc.downloadTool(url)
-  await exec.exec('sudo', ['bash', installerPath, '-s', '-a', '-s', '--action', 'install', '--eula', 'accept'])
-  await io.rmRF(installerPath)
-  await prune()
+    return components
 }
 
-async function installList (components) {
-  for (const component of components) {
-    await install(component)
-  }
+function setVars(installPath, components) {
+    const platform = getPlatform()
+    // is it safe to assume top-level setvars scripts will always exist?
+    const script = platform === 'windows'
+        ? path.join(installPath, 'setvars-vcvarsall.bat')
+        : path.join(installPath, 'setvars.sh')
+    console.log(`Running environment init script: ${script}`)
+    exec.exec(script).then(exitCode => {
```

Comment:
I don't think exec'ing a script will change the environment for future steps. It looks like you need to do `exportVariable`
from this package https://github.com/actions/toolkit/tree/main/packages/core, similar to the `addPath`  below. It will be hard to do this because there are many variables and many packages. Other tools implemented something that sources setvars.sh and then diffs the previous environment to collect a set of variables, then sets the variables. The compiler should be usable with path and LD_LIBRARY_PATH. The libraries would need CPATH, LIBRARY_PATH. I would save this task for later.